src/orange/Translator.hh

```diff
@@ -7,100 +7,128 @@
 //---------------------------------------------------------------------------//
 #pragma once
 
-#include "corecel/cont/Range.hh"
+#include "corecel/cont/Span.hh"
+#include "corecel/math/Algorithms.hh"
+#include "corecel/math/ArrayOperators.hh"
 #include "orange/OrangeTypes.hh"
 
 namespace celeritas
 {
 //---------------------------------------------------------------------------//
 /*!
- * Translate points from a parent's reference frame into the daughter.
+ * Storage implementation for transformations with rotation.
  *
- * The input "translation" is the transformation applied to a daughter universe
- * to place it in the parent. The daughter is a "lower" level compared to the
- * parent.
+ * The input translation and rotation are specified by rotating the daughter's
+ * coordinate system internally, then translating into the parent. The input
+ * rotation must be orthogonal (real and unitary) and will be checked on
+ * construction.
  */
-class TranslatorDown
+class TranslatorStorage
 {
   public:
-    // Construct with the parent-to-daughter translation
-    explicit inline CELER_FUNCTION TranslatorDown(Real3 const& translation);
+    //@{
+    //! \name Type aliases
+    using StorageSpan = Span<const real_type, 3>;
+    using Vec3 = Array<real_type, 3>;
+    //@}
+
+    //! Transform type identifier
+    static CELER_CONSTEXPR_FUNCTION TransformType transform_type()
+    {
+        return TransformType::tl;
+    }
+
+  public:
+    //! Construct with the translation vector
+    explicit CELER_FUNCTION TranslatorStorage(Vec3 const& trans) : tra_(trans)
+    {
+    }
+
+    //! Construct inline from storage
+    explicit CELER_FUNCTION TranslatorStorage(StorageSpan s)
+        : tra_{s[0], s[1], s[2]}
+    {
+    }
 
-    // Translate a single point
-    CELER_FORCEINLINE_FUNCTION Real3 operator()(Real3 const& parent) const;
+    //! Translation vector
+    CELER_FORCEINLINE_FUNCTION Vec3 const& translation() const { return tra_; }
+    //! Get a view to the data for type-deleted storage
+    CELER_FUNCTION StorageSpan data() const { return {&tra_[0], 3}; }
+
+    // Return new storage with the inverse of this transform
+    TranslatorStorage calc_inverse() const;
 
   private:
-    Real3 const& translation_;
+    Vec3 tra_;
```

Comment:
How about `trans_`? An initializer list with`: foo_(foo)` seems standard

Reply:
Where is `rot_`? I don't see it here or on develop.

Reply:
Sorry about that, I started this PR too bare-bones.